flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleDescriptor.java

```diff
@@ -67,4 +68,31 @@ default boolean isUnknown() {
 	 * @return the resource id of the producing task executor if the partition occupies local resources there
 	 */
 	Optional<ResourceID> storesLocalResourcesOn();
+
+	/**
+	 * Return release types supported by Shuffle Service for this partition.
```

Comment:
remove `for this partition`.
`by this shuffle service`.

Reply:
why? if we leave `by this shuffle service` then it might sound like for all partitions but this method returns supported release types for certain partition.

Reply:
There might exist two options:
- One `ShuffleService` implementation might provide different `ShuffleDescriptor` implementations for different partitions with different release types. Then the previous comment makes sense.

- One `ShuffleService` implementation would only have one kind of `ShuffleDescriptor` and provide one enum of release types for all the partitions. Just like atm we only have `NettyShuffleDescriptor` implementation which provides the same release types for all partitions. So my above comment suggestion was based on this. The release type is `ShuffleService` global level suitable for all partitions.

Reply:
There might exist two options:

One ShuffleService implementation might provide different ShuffleDescriptor implementations for different partitions with different release types. Then the previous comment makes sense.

One ShuffleService implementation would only have one kind of ShuffleDescriptor and provide one enum of release types for all the partitions. Just like atm we only have NettyShuffleDescriptor implementation which provides the same release types for all partitions. So my above comment suggestion was based on this. The release type is ShuffleService global level suitable for all partitions.

Reply:
The `NettyShuffleDescriptor` dopes not have the same release types for all partitions. For non-blocking partitions we do not support `ReleaseType.MANUAL`.

Reply:
Yes,  I have no other concerns here now.